### ISSUE TITLE
Allow kiosk mode to be the default

### ISSUE THREAD

**Issue by mavam** [grafana/grafana#17642]
I would like to make the default view mode *kiosk mode*. The use case is that the dashboard starts in as plain as possible (kiosk mode), without the need to add `?kiosk` to the URL or hitting `d k` on the keyboard. I could not find a configuration option in `grafana.ini`.

I'd be also interested in a workaround if that exists.

**Comment by aocenas** [comment]
Yeah that would be a feature request as we do not have any configuration like that at the moment and I don't think there is any sensible workaround right now.

**Comment by marefr** [comment]
If you use a reverse proxy in front of Grafana like nginx or apache you may be able to use url rewrite to make sure to add the required kiosk parameter, but maybe only work for the first full page load of a dashboard?

**Comment by mavam** [comment]
Yeah, the URL rewrite will probably work, but it's a pretty heavy-weight workaround for the problem at hand.

I don't know the Grafana internals, but I was hoping the initial display mode can be lifted into a config option without too much effort.

Alternatively, being able to set it via the API would work as well, which we also use to set the organization-wide default dashboard immediately after deploying.

**Comment by sandromehic** [comment]
+1

Have a couple of use cases where the best thing would be for dashboards to start in kiosk mode.

**Comment by Xeyk** [comment]
+1

**Comment by zhushilu** [comment]
+1

**Comment by quilldata** [comment]
+1

**Comment by herryliq** [comment]
I had a same problem, the use case is when open the domain url with browser, should be displayed by  kiosk mode.

**Comment by evanmn** [comment]
+1. I've heard that something like this, or a read-only mode has not been added because users could work around this, but this would still be useful to have. I have implemented this in openresty, although it would be nice to have in grafana itself.

**Comment by felipenaf** [comment]
+1
It would be a great update.

**Comment by TJChong92** [comment]
+1

**Comment by krakazyabra** [comment]
This is really good feature, you can provide annonymous access to dashboard and don't be afraid, that user will go out of the dashboard.
+1

**Comment by alex88** [comment]
The nginx reverse proxy trick doesn't work, even if the query param is there not having it in the browser doesn't start the kiosk mode (is it checked client side)?

**Comment by DonatoD** [comment]
@marefr Do you have any news? I even asked help here where you can find my configurations. I need the kiosk mode only for the default dashboard, so the first one after login
https://community.grafana.com/t/nginix-redirect-at-kiosk-mode/57067
Thanks a lot

**Comment by uala-gonzalobrandan** [comment]
+1

**Comment by namhoai** [comment]
+1

**Comment by DonatoD** [comment]
At least, could you please provide help so set for example nginx, or others, to rewrite the url adding "& kiosk" at the end? it's over one year I'm trying everything without success

**Comment by centenico** [comment]
+1

**Comment by shaxxx** [comment]
+1

**Comment by MrMoonCH** [comment]
Would love this feature as well

+1

**Comment by Prabesh01** [comment]
+1

**Comment by sant1dom** [comment]
+1

**Comment by DonatoD** [comment]
I see there is quite a bit of interest. If someone found some workaround with nginx, please post it

**Comment by julianf19** [comment]
If you're doing it in an iframe you can do the following:
`var iframe = document.getElementById("grafana-iframe");
iframe.contentWindow.Mousetrap.reset();`

no keystroke will set off any action

**Comment by epgodoy** [comment]
+1

**Comment by jasonmacer** [comment]
+1 to this as well.

**Comment by yasinarabi** [comment]
+1

**Comment by ToMs77130** [comment]
+1, a kiosk mode according to org list would be great

**Comment by C8opmBM** [comment]
+1

**Comment by Zohair-coder** [comment]
+1

**Comment by Silonei-Duarte** [comment]
+1

**Comment by gareth-e** [comment]
+1
